free-2.md

```diff
@@ -1,44 +1,77 @@
 # Free project 2
 
-
 ## The user and a language
 This section describes who the project would serve and why a language might be a
 good way to meet their needs.
 
+Recipes
 
 ### What's the need?
 _What need is met by your idea? Who are you helping? What is that person's
 experience like now? What would their experience be like if you could help 
 them?_
 
+Right now, recipes can range from overly informative ("use a 3.9 inch pan and 
+fry for precisely 2 minutes and 7.3 seconds") to not nearly informative enough
+("fry the food").  This can cause difficulty for the amateur chef - the devil is
+in the details and the details aren't there.  There experience could be improved
+and streamlined if there was a more formal, universal system of specifying how 
+to cook things.
 
 ### Why a language?
 _Why is a DSL appropriate for your user(s)? How does it address the need?_
 
+I would argue that recipes, in general, are alreay a DSL.  They even do a pretty
+good job when they're well written.  My DSL would be a stricter subset of 
+existing recipes designed to create a deterministic (or as close to as possible)
+method of cooking food - ideally a computer could follow these steps exactly and
+produce the same food as a human.
 
 ### Why you?
 _What excites you about this idea? How did you come up with it?_
 
+Living in an apartment and being off of the meal plan this year I've had to get
+much better at feeding myself.  I often look for new, easy recipes online and 
+find that some are much, much better than others.  This is frustrating because
+sometimes the things I **really** want to cook/eat I can't because I don't 
+understand how they're supposed to boe made.
 
 ### Domain
 _Describe the project's domain in five words._
 
+Precise food prep
 
 ### Interface (syntax)
 _How might the user interact with the language? What does programming look 
 like? Why is this the right way to interact with the problem domain?_ 
 
+Writing the recipe shouldn't change much, but the output of a recipe should be
+exact and strictly adhering to the guidelines.  Reading and using the recipe 
+should be unchanged (except that it is easier and more straightforward).  
+Programming is basically cooking, which is what you want to do!
+
 
 ### Operation (semantics)
 _What might happen when a program runs? How does a program interact with the
 user? What kinds of errors might occur, and how might they be communicated to
 the user?_
 
+When handling a written recipe, it should use NLP plus a good knowledge of what
```

Comment:
I don't think you'd necessarily need NLP for this project - that would only be if you wanted to try to parse existing recipes into the language. You could make the syntax of the language as restrictive as you want and force users to conform to it.